kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/CaseServicesClientImpl.java

```diff
@@ -878,22 +878,26 @@ public void removeComment(String containerId, String caseId, String commentId) {
         return getCaseInstances(null, page, pageSize, sort, sortOrder);
     }
 
-    @Override
     public List<CaseInstance> getCaseInstances(List<String> status, Integer page, Integer pageSize, String sort, boolean sortOrder) {
+        return getCaseInstances (status, page, pageSize, sort, sortOrder, false);
+    }
+
+    @Override
+    public List<CaseInstance> getCaseInstances(List<String> status, Integer page, Integer pageSize, String sort, boolean sortOrder, boolean withData) {
         CaseInstanceList list = null;
         if( config.isRest() ) {
             Map<String, Object> valuesMap = new HashMap<String, Object>();
 
             String queryString = getPagingQueryString("", page, pageSize);
             queryString = getAdditionalParams(queryString, "status", status);
             queryString = getSortingQueryString(queryString, sort, sortOrder);
-
+            queryString = getAdditionalParams(queryString, "withData", Collections.singletonList(withData));
             list = makeHttpGetRequestAndCreateCustomResponse(
                     build(loadBalancer.getUrl(), CASE_QUERY_URI + "/" + CASE_ALL_INSTANCES_GET_URI, valuesMap) + queryString, CaseInstanceList.class);
 
         } else {
             CommandScript script = new CommandScript( Collections.singletonList(
-                    (KieServerCommand) new DescriptorCommand("CaseQueryService", "getCaseInstancesAnyRole", new Object[]{safeList(status), page, pageSize, sort, sortOrder})) );
+                    (KieServerCommand) new DescriptorCommand("CaseQueryService", "getCaseInstancesAnyRole", new Object[]{safeList(status), page, pageSize, sort, sortOrder, withData})) );
```

Comment:
Arrays should not be created for varargs parameters
```suggestion
                    (KieServerCommand) new DescriptorCommand("CaseQueryService", "getCaseInstancesAnyRole", safeList(status), page, pageSize, sort, sortOrder, withData)) );
```